pkg/reconciler/kubernetes/initcontroller/initcontroller.go

```diff
@@ -54,24 +57,28 @@ func (ctrl Controller) InitController(ctx context.Context, opts PayloadOptions)
 		ctrl.Logger.Fatalw("Error creating initial manifest", zap.Error(err))
 	}
 
+	operatorVersion := os.Getenv(versionKey)
+	if operatorVersion == "" {
+		ctrl.Logger.Fatal("Failed to find version from env")
```

Comment:
Why do we want to use Fatal Here?

Reply:
yes so we have VERSION var in operator deployment
and if it is not found that means we can't do upgrade conditions
so can't move ahead